.github/workflows/docker.yml

```
@@ -0,0 +1,57 @@
+name: Build and Push Docker Images
+
+on:
+  push:
+    # TODO: run on tags rather than branches
+    branches: [ "main" ]
+    paths:
+      - 'Dockerfile'
+      - 'uc-cli.dockerfile'
+      - '.github/workflows/docker.yml'
+
+  pull_request:
+    branches: [ "main" ]
+    paths:
+      - 'Dockerfile'
+      - 'uc-cli.dockerfile'
+      - '.github/workflows/docker.yml'
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - dockerfile: Dockerfile
+            image: unitycatalog
+          - dockerfile: uc-cli.dockerfile
+            image: uunitycatalog-cli
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Login to Docker Hub
+      uses: docker/login-action@v3
+      with:
+        username: ${{ vars.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+    - name: Build and push Docker image
+      uses: docker/build-push-action@v6
+      with:
+        context: .
+        file: ${{ matrix.dockerfile }}
+        push: true
```

```suggestion
        push: false
```

To be removed; included to demonstrate pushing to my personal DockerHub. This PR should only add builds and any other relevant tests for the images. Pushing and tagging should be decided later.